modules/ROOT/pages/secure-storage.adoc

```diff
@@ -0,0 +1,110 @@
+
+Windows Integration (64 bit)::
+Available on Windows. This provider uses a random value that is encrypted based on your Windows login information and stored in secure storage. This provider uses the master password automatically to decrypt your credentials.
+OS X Keystore Integration::
+Available on macOS. This provider uses a random value that is stored in the OS Keychain. You can check it using the Keychain Access utility. This provider uses the master password automatically to decrypt your credentials.
+Linux Integration (64 bit)::
+Available in Linux. This provider uses a random value and relies on the `libsecret` library that must be present and working on your Linux distribution. This provider uses the master password automatically to decrypt your credentials.
+UI Prompt::
+Available in Windows, macOS, and Linux. This provider prompts you to create a master password and uses it to encrypt your credentials. This provider requires you to enter your master password every time you open Studio.
+
+When adding a new credential for the first time, Studio notifies you that a new master password has been created. Additionally, Studio prompts you to create two password hints to use in case you forget your master password. The password hints are security questions that you must treat as a second password. If you lose your master password, your password hints are the only other method to recover your credentials.
+
+Each provider supplies its own master password. When you use a provider to encrypt a new credential, you can only use that provider to recover the encrypted credential.
+
+Once a master password is created, Studio caches it until it is closed, or you manually clear the password from Studio's cache.
+
+== Secure Storage File
+
+This file holds all your user credentials in an encrypted form. By default, the secure storage file is located in `$HOME/.eclipse/org.eclipse.equinox.security/secure_storage`.
+
+Because it is a single file, the same secure storage is shared across all Eclipse-based workspaces and Anypoint Studio instances.
+
+
+== Update the Master Password
+
+You can change or rotate the master password for a particular provider. When changing a master password, all the encrypted data by this provider is re-encrypted with the new master password.
+
+=== Before you Begin
+
+* Ensure that all other instances of Anypoint Studio and any other Eclipse-based application are not running. +
+Because the secure storage file is shared across all running instances and the master password is cached in each instance's memory, updating your master password causes other opened instances to still write the credentials using the old master password. Closing all other instances ensures you avoid decryption issues. See <<errors, Decryption Errors>> for more information.
```

Comment:
I think of prerequisites as WHAT is required, not WHY, which is conceptual information.
That said, I would make this a one-sentence text:
```suggestion
To avoid decryption issues, ensure that all other instances of Anypoint Studio and any other Eclipse-based application are not running. +
```